projects/kalidea/kaligraphi/src/lib/molecules/kal-dialog/kal-dialog.service.ts

```diff
@@ -0,0 +1,165 @@
+import { Injectable, Injector, Optional } from '@angular/core';
+import { Location } from '@angular/common';
+import { Overlay, OverlayRef } from '@angular/cdk/overlay';
+import { ComponentPortal, ComponentType, PortalInjector } from '@angular/cdk/portal';
+
+import { KAL_DIALOG_DATA } from './kal-dialog.injector';
+import { KalDialogRef } from './kal-dialog-ref';
+import { KalDialogConfig } from './kal-dialog-config';
+import { KalDialogContainerComponent } from './dialog-container/kal-dialog-container.component';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class KalDialogService {
+
+  /**
+   * list of opened dialogs
+   */
+  private dialogsList: KalDialogRef<any, any>[] = [];
+
+  constructor(private overlay: Overlay,
+              private injector: Injector,
+              @Optional() private location?: Location) {
+  }
+
+  /**
+   * get dialogs list
+   */
+  getDialogsList() {
+    return this.dialogsList;
+  }
+
+  /**
+   * open Dialog
+   */
+  open<T, D>(component: ComponentType<T>,
+             config?: KalDialogConfig<D>) {
+
+    config = this.applyConfig(config);
+
+    const overlayRef = this.createOverlay(config);
+    const dialogContainer = this.attachDialogContainer(overlayRef, config);
+    const dialogRef = this.attachDialogContent<T, D>(component,
+      dialogContainer,
+      overlayRef,
+      config);
+
+    dialogContainer.setDialogRef(dialogRef);
+
+    // // register dialog
+    this.dialogsList.push(dialogRef);
+
+    return dialogRef;
+
+  }
+
+  /**
+   * retrieve dialog by ID
+   */
+  getDialog(id: string) {
+    return this.dialogsList.find(dialog => dialog.id === id);
+  }
+
+
+  /**
+   * retrieve dialog by ID
+   */
+  close<T, D>(dialogRef: KalDialogRef<T, D>) {
+    dialogRef.overlayRef.detach();
+  }
+
+  detachOverlay(overlayRef: OverlayRef) {
+    overlayRef.detach();
+  }
+
+  /**
+   * set default config
+   */
+  private applyConfig(config) {
+    // default config
+    const positionStrategy = this.overlay.position().global().centerVertically().centerHorizontally();
+    const scrollStrategy = this.overlay.scrollStrategies.block();
+
+    config = Object.assign(new KalDialogConfig(), {positionStrategy, scrollStrategy}, config);
```

Comment:
You don't have to assign the config again, you can just `return Object.assign....`